### DISCUSSION TITLE
Filter Dictionary of Tags

### DISCUSSION THREAD

**Discussion by chrisegge** [dmroeder/pylogix#206] [Q&A]
How would I filter out a lists of tag names from the Dictionary of Tags that is returned by a specific tag type, in this case a UDT?

**Comment by TheFern2** [comment]
Please post minimal code example of what you're trying to achieve. i.e. I'm writing this X code, and I would like to get Y result.

**Reply by chrisegge** [reply]
I was trying to modify the example -- 

from pylogix import PLC
with PLC("192.168.1.9") as comm:
    tags = comm.GetTagList()
    for t in tags.Value:
        print("Tag:", t.TagName, t.DataType)

and then filter it to do something like print("Tag:", where t.datatype == "ThisMotorUDT"

**Reply by TheFern2** [reply]
You could just do a quick check if the datatype matches whatever you're looking for, and append it to a new list:

Quick example:

```python
from pylogix import PLC

with PLC("192.168.1.14", 1) as comm:
    tags = comm.GetTagList()
    filtered_list = []
    for t in tags.Value:
        print("Tag:", t.TagName, t.DataType)
        if t.DataType == "ThisMotorUDT":
            filtered_list.append(t.TagName)

    print("################################")

    for t in filtered_list:
        print("Tag:", t)
```
